### ISSUE TITLE
Default Java version changed from 7 to 8; missing MacOS JDK version docs

### ISSUE THREAD

**Issue by adamretter** [travis-ci/travis-ci#6716]
It seems the default version of Java has changed on both Linux and MacOS VM's from Java 7 to Java 8, which has broken our builds.

I have specified java: oraclejdk7 which fixes the linux build, but for the Mac build your documentation states: https://docs.travis-ci.com/user/osx-ci-environment/#JDK-and-OS-X but when I look at the images e.g. https://docs.travis-ci.com/user/osx-ci-environment/#Xcode-7.3.1 it doesn't tell me which version of the JDK it comes with, so I have no idea which Mac OS image to use.

It seems there is a hole in the docs. Any ideas?

**Comment by BanzaiMan** [comment]
We currently do not support switching JDK on OS X. https://github.com/travis-ci/travis-ci/issues/2317

If the default JDK version changed, that is due to the changes in OS X itself.

**Comment by adamretter** [comment]
@BanzaiMan Is it possible to know which OSX image has which JDK, your documentation claims that this is the case, but it is missing from the documentation.

**Comment by adamretter** [comment]
@BanzaiMan I guess this is really asking for: https://github.com/travis-ci/docs-travis-ci-com/issues/593

It seems bad to me though that you updated the oraclejdk version in the existing MacOS images without warning users. It seems now that from #593, that you don't provide Java 7 at all on MacOS?

**Comment by trejkaz** [comment]
> If the default JDK version changed, that is due to the changes in OS X itself.

Not really, no, because macOS doesn't ship with a JDK. Which version of Java goes with which version of macOS is entirely chosen by Travis CI.
